src/ec/packages/ec-component-ecl-compliance/components/ecl-compliance-media-container.html.twig

```diff
@@ -0,0 +1,56 @@
+{#
+  Parameters:
+    alt (string),
+    image (string),
+    embedded_media: (string),
+    sources: (array of objects),
+    tracks: (array of objects)
+#}
+
+{% extends "@ecl-twig/ec-component-ecl-compliance/ecl-compliance.html.twig" %}
+
+{% set message = '' %}
+{% set component = 'media-container' %}
+{% set no_embedded_media_case = embedded_media is not defined or embedded_media is empty %}
+{% set no_sources_case = sources is not defined or sources is empty or sources is not iterable %}
+{% set no_tracks_case = tracks is not defined or tracks is empty or tracks is not iterable %}
+{% set no_video_resources_case = no_sources_case and no_tracks_case %}
+{% set no_image_case = image is not defined or image is empty %}
+{% set no_alt_case = no_image_case == false and (alt is not defined or alt is empty) %}
+
+{# no media provided at all #}
+{% set not_compliant = no_image_case and no_embedded_media_case and no_video_resources_case %}
+{% if not_compliant %}
+  {% set message = 'One media is required for the media-container component, an "image", an "embedded_media" or video "tracks" and "sources".' %}
+{% endif %}
+{# No embedded_media, we check for video sources #}
+{% if no_embedded_media_case and no_sources_case == false %}
```

Comment:
same question whether an OR wouldn't be more appropriate in order to maximize catching errors.

Reply:
The comments here were trying to help, we follow the logic in the template, which renders the embedded_media if it's present, then it goes rendering a video with sources and tracks.
In this case the condition needs to check for the embedded_media before even checking for the sources, and only in case the first is not set and the second is, it loops into the sources to spot some missing parameter.
So, no, in this case an OR approach would not catch more errors, it would instead create errors where we don't have, potentially, imagine you set the embedded_media but then you also have a sources array defined, with some problems, the template renders the embedded_media and it's correct, valid, although it contains some additional, unneeded data.
Hope this explains :)

Reply:
got the idea